src/types/utils.cpp

```diff
@@ -428,6 +428,90 @@ catch (...)
     return {};
 }
 
+// Routine Description:
+// - Pre-process text pasted (presumably from the clipboard) with provided option.
+// Arguments:
+// - wstr - String to process.
+// - option - option to use.
+// Return Value:
+// - The result string.
+std::wstring Utils::FilterStringForPaste(const std::wstring_view wstr, const FilterOption option)
+{
+    std::wstring filtered;
+    filtered.reserve(wstr.length());
+
+    const auto isControlCode = [](wchar_t c) {
+        if (c >= L'\x20' && c <= L'\x7f')
+        {
+            // Printable ASCII characters + DEL.
+            return false;
+        }

```

Comment:
Last time I tested this functionality, I believe both XTerm and VTE filtered out the DEL character. At least in some sense - XTerm converted it to space, while VTE filtered it out completely.